hal/halx86/include/smp.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Header File for SMP
+ * PROGRAMMERS: Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+#pragma once
+
+/* This table is filled for each physical processor on system */
+typedef struct _PROCESSOR_IDENTITY
+{
+    UCHAR ProcessorId;
+    UCHAR LapicId;
+    BOOLEAN ProcessorStarted;
+    BOOLEAN BSPCheck;
+    PKPRCB ProcessorPrcb;
+
+} PROCESSOR_IDENTITY, *PPROCESSOR_IDENTITY;
+
+/* This table is counter of the overall APIC constants acquired from madt */
+typedef struct _HALP_APIC_INFO_TABLE
+{
+    ULONG ApicMode;
+    ULONG ProcessorCount; /* Count of all physical cores, This includes BSP */
+    ULONG IOAPICCount;
+    ULONG LocalApicPA;                // The 32-bit physical address at which each processor can access its local interrupt controller
+    ULONG IoApicVA[256];
+    ULONG IoApicPA[256];
+    ULONG IoApicIrqBase[256]; // Global system interrupt base 
+
+} HALP_APIC_INFO_TABLE, *PHALP_APIC_INFO_TABLE;
+
+/* 
+ * when we implement NT6.1+ style processor groups this limit will change
+ * for more information here's a link to geoffs site:
+ * https://www.geoffchappell.com/studies/windows/km/ntoskrnl/api/ke/affinity/index.htm
+ */
+#ifdef WIN64
+#define MAX_CPUS 64
+#elif WIN32
+#define MAX_CPUS 32
+#endif
```

Comment:
There exists already a define called `MAXIMUM_PROCESSORS` (in the DDK) that does precisely that.